SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.cxx

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#define LOCTEXT_NAMESPACE "SpatialConstants"
+
+using namespace SpatialGDK;
+
+namespace SpatialConstants
+{
+
+FString RPCTypeToString(ERPCType RPCType)
+{
+    switch (RPCType)
+    {
+    case ERPCType::ClientReliable:
+        return TEXT("Client, Reliable");
+    case ERPCType::ClientUnreliable:
+        return TEXT("Client, Unreliable");
+    case ERPCType::ServerReliable:
+        return TEXT("Server, Reliable");
+    case ERPCType::ServerUnreliable:
+        return TEXT("Server, Unreliable");
+    case ERPCType::ServerAlwaysWrite:
+        return TEXT("Server, AlwaysWrite");
+    case ERPCType::NetMulticast:
+        return TEXT("Multicast");
+    case ERPCType::CrossServer:
+        return TEXT("CrossServer");
+    }
+
+    checkNoEntry();
+    return FString();
+}
+
+TOptional<ERPCType> RPCStringToType(const FString& String)
+{
+    static const TMap<FString, ERPCType> s_NamesMap = []
+    {
+        TMap<FString, ERPCType> NamesMap;
+        for (uint8 RPCTypeIdx = static_cast<uint8>(ERPCType::RingBufferTypeBegin);
+            RPCTypeIdx <= static_cast<uint8>(ERPCType::RingBufferTypeEnd); RPCTypeIdx++)
+        {
+            ERPCType RPCType = static_cast<ERPCType>(RPCTypeIdx);
+            NamesMap.Add(RPCTypeToString(RPCType), RPCType);
+        }
+        return NamesMap;
+    }();
+
+    const ERPCType* Entry = s_NamesMap.Find(String);
+
+    if (Entry)
+    {
+        return *Entry;
+    }
+
+    return {};
+}
+
+
+const FString SERVER_AUTH_COMPONENT_SET_NAME = TEXT("ServerAuthoritativeComponentSet");
+const FString CLIENT_AUTH_COMPONENT_SET_NAME = TEXT("ClientAuthoritativeComponentSet");
+const FString DATA_COMPONENT_SET_NAME = TEXT("DataComponentSet");
+const FString OWNER_ONLY_COMPONENT_SET_NAME = TEXT("OwnerOnlyComponentSet");
+const FString HANDOVER_COMPONENT_SET_NAME = TEXT("HandoverComponentSet");
+const FString ROUTING_WORKER_COMPONENT_SET_NAME = TEXT("RoutingWorkerComponentSet");
+const FString INITIAL_ONLY_COMPONENT_SET_NAME = TEXT("InitialOnlyComponentSet");
+
+const PhysicalWorkerName TRANSLATOR_UNSET_PHYSICAL_NAME = FString("UnsetWorkerName");
+
+const FName DefaultLayer = FName(TEXT("DefaultLayer"));
+
+const FName RoutingWorkerType(TEXT("RoutingWorker"));
+
+const FString ClientsStayConnectedURLOption = TEXT("clientsStayConnected");
+const FString SpatialSessionIdURLOption = TEXT("spatialSessionId=");
+
+const FString LOCATOR_HOST = TEXT("locator.improbable.io");
```

Comment:
Is it worth changing some of these to be const char* since they are converted from wchar->char to pass into the worker APIs?